silverback/types.py

```diff
@@ -0,0 +1,42 @@
+from typing import Optional, Protocol
+
+from pydantic import BaseModel
+from typing_extensions import Self  # Introduced 3.11
+
+
+class ISilverbackSettings(Protocol):
+    """Loose approximation of silverback.settings.Settings.  If you can, use the class as
+    a type reference."""
+
+    INSTANCE: str
+    PERSISTENCE_CLASS: Optional[str]
+
+    def get_network_choice(self) -> str:
+        ...
+
+
+class SilverbackIdent(BaseModel):
```

Comment:
Would prefer this as more typical
```suggestion
class SilverbackID(BaseModel):
```